### PR TITLE
Feat/round slider 2

### PR DESCRIPTION
**Pull request by wiktoriatomzik**

### CODE REVIEW

File: src/Molecules/Slider/Slider.stories.tsx [hunk 1]
```diff
@@ -26,6 +26,7 @@ export const Default = () => {
           onChange={handleChange}
           value={value}
           {...getSliderProps({ min: 0, max: 100, step: 1 })}
+          sliderColor={t => t.color.pill1}
```
Comment (mjmonline) [inline src/Molecules/Slider/Slider.stories.tsx:29]: There is no default color?

File: src/Molecules/Slider/Slider.tsx [hunk 2]
```diff
@@ -1,82 +1,121 @@
...
   );
-  height: 15px;
+  background: ${p => p.$disabled && p.theme.color.sliderDisabled};
+  ${height};
   max-width: 100%;
   width: 100%;
 `;
 
-const StyledSlider = styled.div`
+const StyledSlider = styled('div')<InternalSliderTypes>`
```
Comment (mjmonline) [inline src/Molecules/Slider/Slider.tsx:64]: why not `styled.div`?

File: src/Molecules/Slider/Slider.tsx [hunk 3]
```diff
@@ -1,82 +1,121 @@
...
   transform: translateY(-50%);
-  background: ${p => p.theme.color.sliderThumbColor};
-  border: 1px solid ${p => p.theme.color.sliderThumbBorder};
+  background: ${p => p.theme.color.sliderThumbBackground};
+  border: ${p =>
+    `${p.theme.spacing.unit(1)}px solid ${
+      p.$sliderColor && !p.$disabled ? p.$sliderColor(p.theme) : p.theme.color.sliderColor
+    }`};
+  border: ${p =>
+    p.$disabled ? `${p.theme.spacing.unit(1)}px solid ${p.theme.color.sliderDisabled}` : ''};
```
Comment (mjmonline) [inline src/Molecules/Slider/Slider.tsx:97]: if it's disabled you will set border to empty here, also it comes after the rule on line 92 so it's more important, bug potential

File: src/Molecules/Slider/Slider.tsx [hunk 4]
```diff
@@ -159,53 +200,65 @@ const Slider: FC<Props> = ({ onChange, value, max, min, step, leftColor, rightCo
...
+        if (typeof onChange === 'function') {
+          onChange(newX);
+        }
       }
     }
   };
 
-  const gradient = {
+  const sliderBackground = {
     background: `linear-gradient(to right, ${
```
Comment (mjmonline) [inline src/Molecules/Slider/Slider.tsx:227]: you have the same styles on `StyledSliderWrapper` any reason?

File: src/Molecules/Slider/Slider.stories.tsx (hunk 1 above)
Comment (wiktoriatomzik) [reply src/Molecules/Slider/Slider.stories.tsx:29]: there is, check line 167

File: src/Molecules/Slider/Slider.tsx (hunk 3 above)
Comment (wiktoriatomzik) [reply src/Molecules/Slider/Slider.tsx:97]: if it is not disabled* and then the previous rule will apply.

File: src/Molecules/Slider/Slider.tsx (hunk 2 above)
Comment (wiktoriatomzik) [reply src/Molecules/Slider/Slider.tsx:64]: I forgot ;_;

File: src/Molecules/Slider/Slider.tsx (hunk 4 above)
Comment (wiktoriatomzik) [reply src/Molecules/Slider/Slider.tsx:227]: the procent of linearGradient is recalculated on every thumb change (with the getPercentage function). if you disable any of them, the styles don't change dynamically (I've tried it)

**Comment by wiktoriatomzik** [comment]
> Your components height is equal to the slider bar height. This makes it tricky to put in a layout because designers adds spacing to surrounding components based on the whole height of the slider (including thumb height). So for example when a developer sees 10px padding in design and tries to apply that he/she will actually just have 2px spacing

@mjmonline, any suggestions on how to solve that?

File: src/Molecules/Slider/Slider.stories.tsx (hunk 1 above)
Comment (mjmonline) [reply src/Molecules/Slider/Slider.stories.tsx:29]: Then in the default story we try to use all default values, so you can remove `sliderColor` from this story

**Comment by mjmonline** [comment]
> > Your components height is equal to the slider bar height. This makes it tricky to put in a layout because designers adds spacing to surrounding components based on the whole height of the slider (including thumb height). So for example when a developer sees 10px padding in design and tries to apply that he/she will actually just have 2px spacing
> 
> @mjmonline, any suggestions on how to solve that?

@wiktoriatomzik you can have margins on `StyledSlider` to make it equal the height of the thumb

**Comment by wiktoriatomzik** [comment]
> > > Your components height is equal to the slider bar height. This makes it tricky to put in a layout because designers adds spacing to surrounding components based on the whole height of the slider (including thumb height). So for example when a developer sees 10px padding in design and tries to apply that he/she will actually just have 2px spacing
> > 
> > 
> > @mjmonline, any suggestions on how to solve that?
> 
> @wiktoriatomzik you can have margins on `StyledSlider` to make it equal the height of the thumb

@mjmonline would love to do it like that. this is the outcome:
image

File: src/Molecules/Slider/Slider.stories.tsx (hunk 1 above)
Comment (wiktoriatomzik) [reply src/Molecules/Slider/Slider.stories.tsx]: done 👌

**Pull request merged**